### DISCUSSION TITLE
Changing the flow for Create Observation

### DISCUSSION THREAD

**Discussion by mo-nathan** [MushroomObserver/mushroom-observer#1894] [Ideas]
I think we agree that the these days, the first thing we should ask for are images.  If images are provided then we should use those to determine date and lat/long if available (with the ability to edit the date, revise the lat/long, say to hide the lat/long).  If multiple images are provided, the user should be able to select the data any of them.

I think that the named Locality should become optional.  I'm not sure what sort of reverse geocoding we can do from lat/long these days.  I know when I looked at it a number of years ago the options were kind of complex and not very intuitive.  If we could determine something like county from lat/long that might make the most sense.  There are definitely hidden complexities here since it needs to work internationally where there may not be a county concept, or you get places like Louisiana where they have parishes instead of counties and if I remember correctly each city is it's own parish.  There are also case where the same city is in multiple counties (e.g., Portland, Oregon) and you have parks that cross state lines (e.g., The Great Smokies).  It's not totally clear to me what of the existing mapping features for observations should stay as they are and which need to be revised.

**Comment by nimmolo** [comment]
I can respond to the reverse geocoding: it is already implemented in the new Stimulus map controller, so we're good there. Then it's down to parsing the string Google returns.

**Comment by mo-nathan** [comment]
I'm curious how the reverse geocoding works.  Can we generally get a county level name?  How does it deal with parks and other less political locations?  Is there somewhere I can play with what it does?

**Reply by nimmolo** [reply]
https://developers.google.com/maps/documentation/javascript/examples/geocoding-reverse

**Reply by mo-nathan** [reply]
Interesting.  Thanks for sharing that.  It's interesting that it returns multiple results.  In the example they give you can get:

results[0].formatted_address: "277 Bedford Ave, Brooklyn, NY 11211, USA"
results[1].formatted_address: "Grand St/Bedford Av, Brooklyn, NY 11211, USA"
results[2].formatted_address: "Williamsburg, Brooklyn, NY, USA"
results[3].formatted_address: "Brooklyn, NY, USA"
results[4].formatted_address: "New York, NY, USA"
results[5].formatted_address: "Brooklyn, NY 11211, USA"
results[6].formatted_address: "Kings County, NY, USA"
results[7].formatted_address: "New York-Northern New Jersey-Long Island, NY-NJ-PA, USA"
results[8].formatted_address: "New York Metropolitan Area, USA"
results[9].formatted_address: "New York, USA"

They also mention that results[].types is interesting, but I'm sure how to quick see examples of what it returns.  The first street address seem a bit creepy.  Not sure exactly what we want.

**Reply by mo-nathan** [reply]
In the context of creating an observation maybe we just want to give folks a drop down.  Not sure what exactly we would do with that string.  Do we create a Location or do try to replace Locations with somethig better.  Seems like a lot of work.
